internal/driver/glfw/window.go

```diff
@@ -443,6 +448,13 @@ func (w *window) Canvas() fyne.Canvas {
 }
 
 func (w *window) closed(viewport *glfw.Window) {
+	viewport.SetShouldClose(false)
+
+	if w.onCloseIntercepted != nil {
```

Comment:
As `Close()` will call `closed()` this will likely cause a loop.
Following the discussion yesterday if this approach is taken the code to actually close the window would need to move into `Close()`, which could be called from here if no intercept hook is registered.

Reply:
ok, missed this idea, sorry. 
there's one more issue on this way. having interception logic in the _closed_ it's i don't see the way to call it from the test

Reply:
Yes, that's true. I left a comment inline about this - if it is to be tested then we might need something new like 'simulateUserCloseRequest' just for the test driver? Not sure if that is really needed or not.